File: 2784.cpp
class Solution {
public:
    bool isGood(vector<int>& nums) {
        unordered_set<int>count;
        for(int i = 0; i < nums.size(); i++) {
            if(nums[i] >= nums.size()) {
                return false;
            }
            if(count.find(nums[i]) != count.end()) {
                if(nums[i] != nums.size() - 1) {
                    return false;
                }
            }
            count.insert(nums[i]);
        }
        return nums.size() - 1 == count.size();
    }
};

File: 2784.py
class Solution:
    def isGood(self, nums: List[int]) -> bool:
        count = set()
        for num in nums:
            if num >= len(nums):
                return False
            if num in count:
                if num != len(nums) - 1:
                    return False
            count.add(num)

        return len(count) == len(nums) - 1